synapsis/src/main/java/com/soen/synapsis/appuser/AppUserController.java

```diff
@@ -34,6 +35,17 @@ public String getAppUser(@PathVariable Long uid, Model model) {
         return "pages/userpage";
     }
 
+    @GetMapping("/search")
+    public String getUsersLikeName(@AuthenticationPrincipal AppUserDetails user, @RequestParam String name, Model model) {
+        if (!AppUser.isUserAuthenticated()) {
```

Comment:
Is there a reason we want to require authentication? If the user is anonymous, just return all users that match the query.

Reply:
1. The linked user story mentions that a user must login before searching for a user.
2. It is obtained so that the search result does not include the person that searched for the result. 

Do you want me to change the behavior?

Reply:
No, this seems good.